src/main/java/org/ow2/proactive/catalog/util/parser/WorkflowParser.java

```diff
@@ -129,6 +142,25 @@ private void addVariableIfNotNullAndModelIfNotEmpty(
         }
         if (checkIfNotNull(name, model) && checkIfNotEmpty(name, model)) {
             keyValueMapBuilder.add(new KeyValueLabelMetadataEntity(name, model, ATTRIBUTE_VARIABLE_MODEL_LABEL));
+            addDependencyOn(keyValueMapBuilder, value, model);
+        }
+    }
+
+    private void addDependencyIfCatalogObjectModelExist(
+            ImmutableList.Builder<KeyValueLabelMetadataEntity> keyValueMapBuilder, TaskVariable taskVariable) {
+        String name = taskVariable.getName();
+        String value = taskVariable.getValue();
+        String model = taskVariable.getModel();
+        if (checkIfNotNull(name, model) && checkIfNotEmpty(name, model)) {
+            addDependencyOn(keyValueMapBuilder, value, model);
+        }
+
+    }
+
+    private void addDependencyOn(ImmutableList.Builder<KeyValueLabelMetadataEntity> keyValueMapBuilder, String value,
```

Comment:
Filtering can be fine in KeyValueLabelMetadataHelper class.
 But do you believe that KeyValueMap can not store duplicated values? Or will not throw an error?
Maybe it's better to add check here for duplicated value before adding the duplicated Entity?

I may be wrong. Please correct me, if it works fine locally with DB on your side.
I'm afraid that with using DB it will give an error for duplicated key, label.
Because of this constraint in KeyValueLabelMetadataEntity:
`@UniqueConstraint(columnNames = { "CATALOGOBJECTREVISION",
                                                                                          "PA_KEY",
                                                                                          "LABEL" })`